lib/src/facts/windows/collection.cc

```diff
@@ -32,13 +33,26 @@ namespace facter { namespace facts {
 
     vector<string> collection::get_external_fact_directories()
     {
-        // Get the user data path
-        TCHAR szPath[MAX_PATH];
-        if (!SUCCEEDED(SHGetFolderPath(NULL, CSIDL_COMMON_APPDATA, NULL, 0, szPath))) {
-            LOG_DEBUG("error finding COMMON_APPDATA: %1%", system_error());
+        if (user::is_admin()) {
+            // Get the common data path
+            TCHAR szPath[MAX_PATH];
+            if (SUCCEEDED(SHGetFolderPath(NULL, CSIDL_COMMON_APPDATA, NULL, 0, szPath))) {
+                path p = path(szPath) / "PuppetLabs" / "facter" / "facts.d";
+                return {p.string()};
+            }
+
+            LOG_WARNING("error finding COMMON_APPDATA, external facts unavailable: %1%", system_error());
+        } else {
+            auto home = user::home_dir();
+            if (!home.empty()) {
+                path p = path(home) / ".puppetlabs" / "opt" / "facter" / "facts.d";
```

Comment:
@peterhuene I didn't follow the "matches the admin path" part?

That is the "correct" path, at least as we reached consensus on. Historically puppet, facter, etc, and ruby ecosystem supports unix style `dot` directories, e.g. `~/.puppet`, etc. The difference now is that there is more of hierarchy within. The reasoning behind this change is documented in PUP-4189.